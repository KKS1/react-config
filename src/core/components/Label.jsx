import React, { Component } from "react";

const Label = ({ schema, label, id }) => {
  return !schema && label ? <label htmlFor={id}>{label}</label> : "";
};

export default Label;
