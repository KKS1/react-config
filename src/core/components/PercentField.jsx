import React, { Component } from "react";
import styled from "styled-components";
import MaskedInput from "./MaskedInput";
import Label from "./Label";

const WrapperDiv = styled.div`
  padding: 0px;
`;
const defaultMaskconfig = {
  rightAlign: true,
  digits: 2,
  max: 100
}

export default class PercentField extends Component {
  static defaultProps = {
    label: "Label",
    placeholder: "",
    name: "percentFormField",
    onChange: () => {}
  };

  changeHandler = e => {
    this.props.onChange(e.target.value);
  };

  render() {
    const { placeholder, name, value, id, options } = this.props;
    const { maskconfig } = options;

    const inputMaskconfig = {
      ...defaultMaskconfig,
      ...maskconfig
    };

    return (
      <WrapperDiv className="c-label-field">
        <Label {...this.props} />
        <div className="d-flex">
          <MaskedInput
            id={id}
            className="c-label-field__input c-label-field__input--icon-suffixed"
            type="text"
            inputMode="numeric"
            placeholder={placeholder}
            name={name}
            maskconfig={inputMaskconfig}
            value={value == null ? 0 : value}
            onChange={this.changeHandler}
          />
          <i className="far fa-percent c-label-field__icon c-label-field__icon--append" />
        </div>
      </WrapperDiv>
    );
  }
}
