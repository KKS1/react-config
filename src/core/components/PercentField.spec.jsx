import React, { Component } from "react";
import Label from "./Label";
import PercentField from "./PercentField";

describe("PercentField", () => {
  test("render_PercentField", () => {
    const props = {
      id: "formOne_labelOne",
      options: {
        maskconfig: {
          min: 0,
          max: 100,
          digits: 2
        }
      }
    };
    const wrapper = shallow(<PercentField {...props} />);
    const instance = wrapper.instance();
    expect(instance).toMatchSnapshot();
  });

  test("call_onChange_prop", () => {
    const changeHandler = jest.fn();

    const props = {
      id: "formOne_labelOne",
      value: 100,
      options: {
        maskconfig: {
          min: 0,
          max: 100,
          digits: 2
        }
      },
      onChange: changeHandler
    };
    const wrapper = shallow(<PercentField {...props} />);
    const instance = wrapper.instance();
    expect(instance).toMatchSnapshot();

    instance.changeHandler({ target: { value: 80 } });

    expect(changeHandler).toHaveBeenCalledWith(80);
  });
});
