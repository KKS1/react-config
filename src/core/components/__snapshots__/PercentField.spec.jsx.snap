// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PercentField call_onChange_prop 1`] = `
PercentField {
  "changeHandler": [Function],
  "context": Object {},
  "props": Object {
    "id": "formOne_labelOne",
    "label": "Label",
    "name": "percentFormField",
    "onChange": [MockFunction],
    "options": Object {
      "maskconfig": Object {
        "digits": 2,
        "max": 100,
        "min": 0,
      },
    },
    "placeholder": "",
    "value": 100,
  },
  "refs": Object {},
  "setState": [Function],
  "state": null,
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_didScheduleRenderPhaseUpdate": false,
      "_dispatcher": Object {
        "readContext": [Function],
        "useCallback": [Function],
        "useContext": [Function],
        "useDebugValue": [Function],
        "useEffect": [Function],
        "useImperativeHandle": [Function],
        "useLayoutEffect": [Function],
        "useMemo": [Function],
        "useReducer": [Function],
        "useRef": [Function],
        "useState": [Function],
      },
      "_element": <PercentField
        id="formOne_labelOne"
        label="Label"
        name="percentFormField"
        onChange={[MockFunction]}
        options={
          Object {
            "maskconfig": Object {
              "digits": 2,
              "max": 100,
              "min": 0,
            },
          }
        }
        placeholder=""
        value={100}
      />,
      "_firstWorkInProgressHook": null,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_isReRender": false,
      "_newState": null,
      "_numberOfReRenders": 0,
      "_renderPhaseUpdates": null,
      "_rendered": <ForwardRef
        className="c-label-field"
      >
        <Label
          id="formOne_labelOne"
          label="Label"
          name="percentFormField"
          onChange={[MockFunction]}
          options={
            Object {
              "maskconfig": Object {
                "digits": 2,
                "max": 100,
                "min": 0,
              },
            }
          }
          placeholder=""
          value={100}
        />
        <div
          className="d-flex"
        >
          <MaskedInput
            className="c-label-field__input c-label-field__input--icon-suffixed"
            id="formOne_labelOne"
            inputMode="numeric"
            maskconfig={
              Object {
                "digits": 2,
                "max": 100,
                "min": 0,
                "rightAlign": true,
              }
            }
            name="percentFormField"
            onChange={[Function]}
            placeholder=""
            type="text"
            value={100}
          />
          <i
            className="far fa-percent c-label-field__icon c-label-field__icon--append"
          />
        </div>
      </ForwardRef>,
      "_rendering": false,
      "_updater": [Circular],
      "_workInProgressHook": null,
    },
  },
}
`;

exports[`PercentField render_PercentField 1`] = `
PercentField {
  "changeHandler": [Function],
  "context": Object {},
  "props": Object {
    "id": "formOne_labelOne",
    "label": "Label",
    "name": "percentFormField",
    "onChange": [Function],
    "options": Object {
      "maskconfig": Object {
        "digits": 2,
        "max": 100,
        "min": 0,
      },
    },
    "placeholder": "",
  },
  "refs": Object {},
  "setState": [Function],
  "state": null,
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_didScheduleRenderPhaseUpdate": false,
      "_dispatcher": Object {
        "readContext": [Function],
        "useCallback": [Function],
        "useContext": [Function],
        "useDebugValue": [Function],
        "useEffect": [Function],
        "useImperativeHandle": [Function],
        "useLayoutEffect": [Function],
        "useMemo": [Function],
        "useReducer": [Function],
        "useRef": [Function],
        "useState": [Function],
      },
      "_element": <PercentField
        id="formOne_labelOne"
        label="Label"
        name="percentFormField"
        onChange={[Function]}
        options={
          Object {
            "maskconfig": Object {
              "digits": 2,
              "max": 100,
              "min": 0,
            },
          }
        }
        placeholder=""
      />,
      "_firstWorkInProgressHook": null,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_isReRender": false,
      "_newState": null,
      "_numberOfReRenders": 0,
      "_renderPhaseUpdates": null,
      "_rendered": <ForwardRef
        className="c-label-field"
      >
        <Label
          id="formOne_labelOne"
          label="Label"
          name="percentFormField"
          onChange={[Function]}
          options={
            Object {
              "maskconfig": Object {
                "digits": 2,
                "max": 100,
                "min": 0,
              },
            }
          }
          placeholder=""
        />
        <div
          className="d-flex"
        >
          <MaskedInput
            className="c-label-field__input c-label-field__input--icon-suffixed"
            id="formOne_labelOne"
            inputMode="numeric"
            maskconfig={
              Object {
                "digits": 2,
                "max": 100,
                "min": 0,
                "rightAlign": true,
              }
            }
            name="percentFormField"
            onChange={[Function]}
            placeholder=""
            type="text"
            value={0}
          />
          <i
            className="far fa-percent c-label-field__icon c-label-field__icon--append"
          />
        </div>
      </ForwardRef>,
      "_rendering": false,
      "_updater": [Circular],
      "_workInProgressHook": null,
    },
  },
}
`;
